<ui version="4.0" >
 <class>QProjectMConfigDialog</class>
 <widget class="QDialog" name="QProjectMConfigDialog" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>455</width>
    <height>310</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>projectM Settings</string>
  </property>
  <widget class="QLabel" name="label" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>35</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize" >
    <size>
     <width>50</width>
     <height>0</height>
    </size>
   </property>
   <property name="text" >
    <string>Menu Font Path</string>
   </property>
  </widget>
  <widget class="QToolButton" name="menuFontPathToolButton" >
   <property name="geometry" >
    <rect>
     <x>423</x>
     <y>35</y>
     <width>24</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text" >
    <string>...</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="menuFontPathLineEdit" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>35</y>
     <width>287</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
     <horstretch>2</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This changes the font projectM will use for its opengl rendered menus&lt;/p>&lt;/body>&lt;/html></string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>5</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize" >
    <size>
     <width>50</width>
     <height>0</height>
    </size>
   </property>
   <property name="text" >
    <string>Startup Playlist</string>
   </property>
  </widget>
  <widget class="QToolButton" name="startupPlaylistToolButton" >
   <property name="geometry" >
    <rect>
     <x>423</x>
     <y>5</y>
     <width>24</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text" >
    <string>...</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="startupPlaylistLineEdit" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>5</y>
     <width>287</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
     <horstretch>2</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This sets the default preset playlist when qprojectM starts up. This can be either a directory path or preset playlist file. If unspecified then projectM uses a default preset directory&lt;/p>&lt;/body>&lt;/html></string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>95</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text" >
    <string>Preset Duration (s)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="presetDurationSpinBox" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>95</y>
     <width>61</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">Sets how long a preset should be played in seconds before projectM automatically switches to another in the playlist&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>100000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_4" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>95</y>
     <width>171</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text" >
    <string>Smooth Preset Duration (s)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="smoothPresetDurationSpinBox" >
   <property name="geometry" >
    <rect>
     <x>374</x>
     <y>95</y>
     <width>73</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">Sets how long a transition occurs in seconds when smooth preset switching&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>1000000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_5" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>229</y>
     <width>81</width>
     <height>37</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Window Size</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>125</y>
     <width>171</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Maximum FPS</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="maxFPSSpinBox" >
   <property name="geometry" >
    <rect>
     <x>374</x>
     <y>125</y>
     <width>73</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
     <horstretch>2</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This parameter, if nonzero, limits the number of frames per second projectM will attempt to render to screen. 35 is a reasonable limit.&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>0</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="windowWidthSpinBox" >
   <property name="geometry" >
    <rect>
     <x>284</x>
     <y>235</y>
     <width>54</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This modifies the horizontal component of the startup window size &lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="whatsThis" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This modifies the horizontal component of the startup window size &lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>10000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_10" >
   <property name="geometry" >
    <rect>
     <x>344</x>
     <y>185</y>
     <width>43</width>
     <height>38</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>by</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>155</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Texture Size</string>
   </property>
  </widget>
  <widget class="QComboBox" name="textureSizeComboBox" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>155</y>
     <width>61</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
     <horstretch>1</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">Sets the size of the texture. Higher is better, but slower.&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="sizeAdjustPolicy" >
    <enum>QComboBox::AdjustToMinimumContentsLength</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_8" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>185</y>
     <width>81</width>
     <height>38</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Mesh Size</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="fullscreenOnStartupCheckBox" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>243</y>
     <width>186</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Fullscreen on Startup</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="menuOnStartupCheckBox" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>214</y>
     <width>186</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Show Menu on Startup</string>
   </property>
  </widget>
  <widget class="QDialogButtonBox" name="buttonBox" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>272</y>
     <width>250</width>
     <height>30</height>
    </rect>
   </property>
   <property name="layoutDirection" >
    <enum>Qt::RightToLeft</enum>
   </property>
   <property name="orientation" >
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons" >
    <set>QDialogButtonBox::Apply|QDialogButtonBox::Close|QDialogButtonBox::NoButton|QDialogButtonBox::Reset</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="windowHeightSpinBox" >
   <property name="geometry" >
    <rect>
     <x>393</x>
     <y>235</y>
     <width>54</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;">This modifies the vertical component of the startup window size&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>10000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_11" >
   <property name="geometry" >
    <rect>
     <x>344</x>
     <y>229</y>
     <width>43</width>
     <height>37</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>by</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="useAspectCorrectionCheckBox" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>185</y>
     <width>186</width>
     <height>23</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">This makes a square look like a square, and a circle look like a circle in cases where they may not otherwise.&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="text" >
    <string>Use Aspect Correction</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>125</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Beat Sensitivity</string>
   </property>
  </widget>
  <widget class="QToolButton" name="titleFontPathToolButton" >
   <property name="geometry" >
    <rect>
     <x>423</x>
     <y>65</y>
     <width>24</width>
     <height>24</height>
    </rect>
   </property>
   <property name="text" >
    <string>...</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_9" >
   <property name="geometry" >
    <rect>
     <x>5</x>
     <y>65</y>
     <width>119</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize" >
    <size>
     <width>50</width>
     <height>0</height>
    </size>
   </property>
   <property name="text" >
    <string>Title Font Path</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="titleFontPathLineEdit" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>65</y>
     <width>287</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
     <horstretch>2</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">Changes the font projectM will use for rendering the song title&lt;/p>&lt;/body>&lt;/html></string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="beatSensitivitySpinBox" >
   <property name="geometry" >
    <rect>
     <x>130</x>
     <y>125</y>
     <width>61</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">Sets the intensity threshold for the preset hard cuts&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <double>0.000000000000000</double>
   </property>
   <property name="maximum" >
    <double>100.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_13" >
   <property name="geometry" >
    <rect>
     <x>197</x>
     <y>155</y>
     <width>171</width>
     <height>24</height>
    </rect>
   </property>
   <property name="sizePolicy" >
    <sizepolicy vsizetype="Preferred" hsizetype="MinimumExpanding" >
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text" >
    <string>Easter Egg Parameter</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="beatSensitivitySpinBox_2" >
   <property name="geometry" >
    <rect>
     <x>374</x>
     <y>155</y>
     <width>73</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">A mystery parameter!&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="maximum" >
    <double>1.000000000000000</double>
   </property>
   <property name="singleStep" >
    <double>0.010000000000000</double>
   </property>
  </widget>
  <widget class="QSpinBox" name="meshSizeWidthSpinBox" >
   <property name="geometry" >
    <rect>
     <x>284</x>
     <y>192</y>
     <width>54</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;">This modifies the horizontal component of the mesh. Higher is better but slower&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>1000</number>
   </property>
   <property name="value" >
    <number>1</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="meshSizeHeightSpinBox" >
   <property name="geometry" >
    <rect>
     <x>393</x>
     <y>192</y>
     <width>54</width>
     <height>24</height>
    </rect>
   </property>
   <property name="toolTip" >
    <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
p, li { white-space: pre-wrap; }
&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;">This modifies the vertical component of the mesh. Higher is better but slower&lt;/p>&lt;/body>&lt;/html></string>
   </property>
   <property name="minimum" >
    <number>1</number>
   </property>
   <property name="maximum" >
    <number>1000</number>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
